docs/docs/gatsby-lifecycle-apis.md

```diff
@@ -23,37 +23,33 @@ The following model gives a conceptual overview of how data is sourced and trans
-- initializes its cache (stored in `/.cache`)
-- pulls in and preprocesses data ("source and transform nodes") into a GraphQL schema
-- creates pages in memory
-  - from your `/pages` folder
-  - from your `gatsby-node.js` if you implement `createPages`/`createPagesStatefully` (e.g. templates)
-  - from any plugins that implement `createPages`/`createPagesStatefully`
-- extracts, runs, and replaces graphql queries for pages and `StaticQuery`s
-- writes out the pages to cache
+During the main bootstrap sequence Gatsby (in this order):
+
+- reads and validates `gatsby-config.js` to load in your list of plugins (it doesn't run them yet).
+- deletes HTML and CSS files from the previous build (public folder)
+- initializes its cache (stored in `/.cache`) and checks if any plugins have been updated since the last run, if so it deletes the cache
+- runs the plugins
+- starts main bootstrap process
+  - runs [onPreBootstrap](/docs/node-apis/#onPreBootstrap). e.g. implemented by [`gatsby-plugin-typography`](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-typography/src/gatsby-node.js)
+- runs [sourceNodes](/docs/node-apis/#sourceNodes) e.g. implemented by [`gatsby-source-wikipedia`](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-source-wikipedia/src/gatsby-node.js)
+  - within this `createNode` can be called multiple times, which then triggers [onCreateNode](/docs/node-apis/#onCreateNode)
+- creates the first GraphQL Schema
+- runs [resolvableExtensions](/docs/node-apis/#resolvableExtensions) for filetype/language extensions e.g. [`gatsby-plugin-typescript`](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-typescript/src/gatsby-node.js)
+- runs [createPages](/docs/node-apis/#createPages) from the gatsby-node.js in the root directory of the project e.g. implemented by [`page-hot-reloader`](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/bootstrap/page-hot-reloader.js)
+  - within this `createPage` can be called any number of times, which then triggers [onCreatePage](/docs/node-apis/#onCreatePage)
+- runs [createPagesStatefully](/docs/node-apis/#createPagesStatefully)
+- runs source nodes again and it updates the GraphQL schema since pages can include queries to other data. Notice this is the second time source nodes run due to a "chicken and the egg problem". We need the pages to extract the queries but we need to create the schema first which requires to source nodes
+- runs [onPreExtractQueries](/docs/node-apis/#onPreExtractQueries) e.g. implemented by [`gatsby-transformer-sharp`](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-transformer-sharp/src/gatsby-node.js) and [`gatsby-source-contentful`](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-source-contentful/src/gatsby-node.js), and extracts queries from pages and components (`StaticQuery`)
+- compiles queries using a Relay compiler and creates the AST
```

Comment:
```suggestion
- compiles queries using a Relay compiler and creates the Abstract Syntax Tree (AST)
```

Reply:
We recently [replaced Relay compiler with our own query compiler](https://github.com/gatsbyjs/gatsby/pull/19665), so this could be changed to:

```suggestion
- compiles GraphQL queries and creates the Abstract Syntax Tree (AST)
```